mwclient/client.py

```diff
@@ -137,18 +161,18 @@ def __init__(self, host, path='/w/', ext='.php', pool=None, retry_timeout=30,
         self.sleepers = Sleepers(max_retries, retry_timeout, wait_callback)
 
         # Site properties
-        self.blocked = False    # Whether current user is blocked
+        self.blocked = False  # type: Union[Tuple[str, str], bool]  # Is user blocked?
         self.hasmsg = False  # Whether current user has new messages
-        self.groups = []    # Groups current user belongs to
-        self.rights = []    # Rights current user has
-        self.tokens = {}    # Edit tokens of the current user
-        self.version = None
+        self.groups = []  # type: List[str]  # Groups current user is in
+        self.rights = []  # type: List[str]  # Rights current user has
+        self.tokens = {}  # type: Dict[str, str]  # Edit tokens of the current user
+        self.version = None  # type: Union[Tuple[int, ...], Tuple[int, str], None]
```

Comment:
I'm not sure about this one. After playing around with the code that produces this for a bit, the rules seem to be: it will always a tuple, it will be *at least* two items long, and the first two items will be ints. After that, any given item may be a string or an int. E.g. `mwclient.Site.version_tuple_from_generator("MediaWiki 1.5.2a.1.3b.1")` gives `(1, 5, 2, 'a', 1, 3, 'b', 1)`. So I think what we want here is something like `Optional[Tuple[Union[int, str], ...]]`, if that's valid?

Reply:
Yeah seems like you're right, that function is _weird_. I've added a new type `VersionTuple = Tuple[Union[int, str], ...]` to the `types.py` that is used in both locations.